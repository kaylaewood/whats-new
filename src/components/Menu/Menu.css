.menu {
  background-color: #333332;
  display: flex;
  flex-direction: column;
  grid-area: menu;
  height: 100%;
  margin-top: 5em;
  position: fixed;
  width: 18.75em;
}

.menu-button {
  align-items: center;
  background-color: white;
  border: none;
  border-radius: 0 20px 20px 0;
  color: #2F403E;
  display: flex;
  font-size: 1em;
  margin-top: 20px;
  padding: 10px;
  padding-left: 30px;
}

.menu-button:hover {
  background-color: gray;
  color: white;
  font-weight: bold;
}

.clicked-button {
  background-color: #EC5449;
  color: white;
  font-weight: bold;
  transform: scale(1.2);
}

.icon {
  margin-right: 10px;
  width: 25px;
}

@media only screen and (max-width: 740px) {
  .menu {
    height: 100px;
    flex-direction: row;
    justify-content: space-around;
    width: 100%;
  }

  .button-label {
    display: none;
  }

  .menu-button {
    border-radius: 0 0 20px 20px;
    height: 70%;
    width: 18%;
    padding: 0;
    justify-content: center;
    align-items: center;
  }

  .icon {
    margin: 0;
  }

  .clicked-button,
  .menu-button:hover {
    transform: scale(1);
  }
}
