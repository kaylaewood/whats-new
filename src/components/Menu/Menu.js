import React from 'react';
import './Menu.css';
import placeholder from '../../images/placeholder.svg';
import flask from '../../images/flask.svg';
import heart from '../../images/heart.svg';
import projectManagement from '../../images/project-management.svg';
import camera from '../../images/video-camera.svg';
import classNames from 'classnames';
import PropTypes from 'prop-types';

function Menu(props) {
  let clickedLocal = classNames({
    'clicked-button': props.state === props.local
  })
  let clickedTech = classNames({
    'clicked-button': props.state === props.technology
  })
  let clickedEntertainment = classNames({
    'clicked-button': props.state === props.entertainment
  })
  let clickedScience = classNames({
    'clicked-button': props.state === props.science
  })
  let clickedHealth = classNames({
    'clicked-button': props.state === props.health
  })

  return (
    <div className='menu'>
      <button className={`${clickedLocal} menu-button local-button`} onClick={() => props.changeNewsSource(props.local)}>
        <img className='icon' src={placeholder} alt='local news icon' />
        <p className='button-label'>LOCAL NEWS</p>
      </button>
      <button className={`${clickedTech} menu-button technology-button`} onClick={() => props.changeNewsSource(props.technology)}>
        <img className='icon' src={projectManagement} alt='technology icon'/>
        <p className='button-label'>TECHNOLOGY</p>
      </button>
      <button className={`${clickedEntertainment} menu-button entertainment-button`} onClick={() => props.changeNewsSource(props.entertainment)}>
        <img className='icon' src={camera} alt='entertainment icon'/>
        <p className='button-label'>ENTERTAINMENT</p>
      </button>
      <button className={`${clickedScience} menu-button science-button`} onClick={() => props.changeNewsSource(props.science)}>
        <img className='icon' src={flask} alt='science icon'/>
        <p className='button-label'>SCIENCE</p>
      </button>
      <button className={`${clickedHealth} menu-button health-button`} onClick={() => props.changeNewsSource(props.health)}>
        <img className='icon' src={heart} alt='health icon'/>
        <p className='button-label'>HEALTH</p>
      </button>
    </div>
  )
}

Menu.propTypes = {
  local: PropTypes.array,
  health: PropTypes.array,
  science: PropTypes.array,
  technology: PropTypes.array,
  entertainment: PropTypes.array,
  changeNewsSource: PropTypes.func,
  state: PropTypes.string
}

export default Menu;
