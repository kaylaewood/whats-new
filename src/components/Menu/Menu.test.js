import React from 'react';
import ReactDOM from 'react-dom';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow(<Menu
      changeNewsSource = {jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should call changeNewsSource on button click', () => {
    const mockChangeNewsSource = jest.fn();
    const wrapper = shallow(<Menu
      changeNewsSource = {mockChangeNewsSource}
      local = 'local'
      technology = 'tech'
      entertainment = 'entertainment'
      science = 'science'
      health = 'health'
    />)

    wrapper.find('.local-button').simulate('click');
    expect(mockChangeNewsSource).toHaveBeenCalledWith('local');

    wrapper.find('.technology-button').simulate('click');
    expect(mockChangeNewsSource).toHaveBeenCalledWith('tech');

    wrapper.find('.entertainment-button').simulate('click');
    expect(mockChangeNewsSource).toHaveBeenCalledWith('entertainment');

    wrapper.find('.science-button').simulate('click');
    expect(mockChangeNewsSource).toHaveBeenCalledWith('science');

    wrapper.find('.health-button').simulate('click');
    expect(mockChangeNewsSource).toHaveBeenCalledWith('health');
  })
});
