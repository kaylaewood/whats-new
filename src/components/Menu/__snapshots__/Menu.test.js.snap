// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should correctly match the snapshot with all the data passed correctly 1`] = `
<div
  className="menu"
>
  <button
    className="clicked-button menu-button local-button"
    onClick={[Function]}
  >
    <img
      alt="local news icon"
      className="icon"
      src="placeholder.svg"
    />
    <p
      className="button-label"
    >
      LOCAL NEWS
    </p>
  </button>
  <button
    className="clicked-button menu-button technology-button"
    onClick={[Function]}
  >
    <img
      alt="technology icon"
      className="icon"
      src="project-management.svg"
    />
    <p
      className="button-label"
    >
      TECHNOLOGY
    </p>
  </button>
  <button
    className="clicked-button menu-button entertainment-button"
    onClick={[Function]}
  >
    <img
      alt="entertainment icon"
      className="icon"
      src="video-camera.svg"
    />
    <p
      className="button-label"
    >
      ENTERTAINMENT
    </p>
  </button>
  <button
    className="clicked-button menu-button science-button"
    onClick={[Function]}
  >
    <img
      alt="science icon"
      className="icon"
      src="flask.svg"
    />
    <p
      className="button-label"
    >
      SCIENCE
    </p>
  </button>
  <button
    className="clicked-button menu-button health-button"
    onClick={[Function]}
  >
    <img
      alt="health icon"
      className="icon"
      src="heart.svg"
    />
    <p
      className="button-label"
    >
      HEALTH
    </p>
  </button>
</div>
`;
